ql/src/java/org/apache/hadoop/hive/ql/exec/tez/TezTask.java

```diff
@@ -688,6 +689,13 @@ DAGClient submit(DAG dag, Ref<TezSessionState> sessionStateRef) throws Exception
     return new SyncDagClient(dagClient);
   }
 
+  private DAGClient submitInternal(DAG dag, TezSessionState sessionState) throws TezException, IOException {
+    TezClient tezClient = sessionState.getSession();
+    DAGClient dagClient = tezClient.submitDAG(dag);
+    runtimeContext.init(tezClient);
```

Comment:
what does it do, look like a post hook or something?

Reply:
this is the initialization of the RuntimeContext that I introduced in HIVE-28623...when the DAG is submitted, we have the info for storing runtime information in that context object

Reply:
can we refactor
````
private DAGClient submitInternal(DAG dag, TezSessionState sessionState) throws TezException, IOException {
  runtimeContext.init(sessionState);
  return sessionState.getSession().submitDAG(dag);
}

TezRuntimeContext.class
public void init(TezSessionState sessionState) {
  this.amAddress = sessionState.getAppMasterUri();
}

TezSessionState.class
String getAppMasterUri() {
  return Optional.of(getSession()).map(
      session -> session.getAmHost() + ":" + session.getAmPort())
    .get();
}
````